packages/devtools_app/lib/src/timeline/timeline_controller.dart

```diff
@@ -231,8 +283,14 @@ class FrameBasedTimeline {
 
   Stream<TimelineFrame> get onSelectedFrame => _selectedFrameController.stream;
 
-  Future<double> get displayRefreshRate async =>
-      data?.displayRefreshRate ?? await serviceManager.getDisplayRefreshRate();
+  Future<double> get displayRefreshRate async {
+    final refreshRate =
+        await serviceManager.getDisplayRefreshRate() ?? defaultRefreshRate;
+    data?.displayRefreshRate = refreshRate;
+    return refreshRate;
+  }
+
+  double get displayRefreshRateCached => data?.displayRefreshRate;
```

Comment:
it would be cleaner to not have a separate
displayRefreshRateCached method.
I assume this is offline related and what you really need is to 
have a faked out serviceManager in offline mode that always returns the default refresh rate or to 
have a conditional in the displayRefreshRate method written above.